packages/manager/apps/catalog/src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { OsdsSearchBar } from '@ovhcloud/ods-components/search-bar/react';
+import { OsdsButton } from '@ovhcloud/ods-components/button/react';
+import { OsdsIcon } from '@ovhcloud/ods-components/icon/react';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+import { useTranslation } from 'react-i18next';
+import { ODS_ICON_NAME, ODS_ICON_SIZE } from '@ovhcloud/ods-components/icon';
+import {
+  ODS_BUTTON_VARIANT,
+  ODS_BUTTON_SIZE,
+  ODS_BUTTON_TYPE,
+} from '@ovhcloud/ods-components/button';
+import Filters from '@/components/Filters/Filters';
+import { Product } from '@/utils/utils';
+
+interface SearchbarProps {
+  products: Product[];
+  setSelectedCategories: React.Dispatch<React.SetStateAction<string[]>>;
+  setSelectedUniverses: React.Dispatch<React.SetStateAction<string[]>>;
+  setSearchValue: React.Dispatch<React.SetStateAction<string>>;
+}
+
+const SearchBar: React.FC<SearchbarProps> = ({
+  products,
+  setSelectedCategories,
```

Comment:
the React way is not to pass a "set" function as a prop but rather the component emitting an "onSearch" event that you will handle in your upper scope

Reply:
we are following this way https://react.dev/learn/thinking-in-react#step-5-add-inverse-data-flow
when we update the search bar text, or checkboxes, we have to pass the states in the hook useCatalog to update the product.